src/main/kotlin/me/partlysanestudios/partlysaneskies/features/themes/ThemeManager.kt

```diff
@@ -0,0 +1,348 @@
+//
+// Written by Su386.
+// See LICENSE for copyright and license notices.
+//
+package me.partlysanestudios.partlysaneskies.features.themes
+
+import cc.polyfrost.oneconfig.config.core.OneColor
+import gg.essential.elementa.components.UIImage
+import gg.essential.universal.utils.ReleasedDynamicTexture
+import me.partlysanestudios.partlysaneskies.PartlySaneSkies.Companion.config
+import me.partlysanestudios.partlysaneskies.PartlySaneSkies.Companion.minecraft
+import me.partlysanestudios.partlysaneskies.utils.ElementaUtils.uiImageFromResourceLocation
+import me.partlysanestudios.partlysaneskies.utils.ImageUtils.loadImage
+import me.partlysanestudios.partlysaneskies.utils.ImageUtils.replaceColor
+import me.partlysanestudios.partlysaneskies.utils.ImageUtils.saveImage
+import net.minecraft.client.renderer.texture.TextureUtil
+import net.minecraft.client.resources.IResource
+import net.minecraft.util.ResourceLocation
+import java.awt.Color
+import java.awt.image.BufferedImage
+import java.io.File
+import java.io.IOException
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+
+object ThemeManager {
+    private val PRIMARY_DEBUG_COLOR: Color = Color(255, 0, 0)
+    private val SECONDARY_DEBUG_COLOR: Color = Color(0, 0, 255)
+    private val ACCENT_DEBUG_COLOR: Color = Color(0, 255, 0)
+    private var lastThemeName: String = ""
+    private val backgroundUIImages: ArrayList<UIImage> = ArrayList()
+    private val buttonDataList: ArrayList<ButtonData> = ArrayList()
+    private val toggleDataList: ArrayList<ToggleData> = ArrayList()
+    private val defaultThemes: Array<Theme> = arrayOf(
+        Theme("Classic (Dark)", Color(46, 47, 50), Color(37, 38, 41), Color(1, 255, 255)),
```

Comment:
You could maybe make this an enum

Reply:
I use the theme constructor for custom themes I beleive

Reply:
I'll fix it when I bring resizable background images